dateparser/custom_language_detection/langdetect.py

```diff
@@ -0,0 +1,38 @@
+from langdetect.detector_factory import DetectorFactory, PROFILES_DIRECTORY
+import langdetect
+
+
+# The below _Factory is set to prevent setting global state of the library
+# but still get consistent results.
+# Refer - https://github.com/Mimino666/langdetect
+
+class _Factory:
+    data = None
+
+
+def _init_factory():
+    if _Factory.data is None:
+        _Factory.data = DetectorFactory()
+        _Factory.data.load_profile(PROFILES_DIRECTORY)
+        _Factory.data.seed = 0
+
+
+def _get_language_probablities(text):
+    _init_factory()
+    detector = _Factory.data.create()
+    detector.append(text)
+    return detector.get_probabilities()
```

Comment:
Maybe I'm missing something, but why are we using a class instead of using directly a variable / object?

What about doing:
```suggestion
_factory_data = None


def _init_factory():
    if _factory_data is None:
        _factory_data = DetectorFactory()
        _factory_data.load_profile(PROFILES_DIRECTORY)
        _factory_data.seed = 0


def _get_language_probablities(text):
    _init_factory()
    detector = _factory_data.create()
    detector.append(text)
    return detector.get_probabilities()
```

If it's possible I would use this approach, I think is simpler.

Reply:
Hi @noviluni, I tried changing class to function caching variable but it throws `UnboundLocalError: local variable '_factory_data' referenced before assignment `

No changes were made